Library/Client.ts

```diff
@@ -234,10 +234,11 @@ class Client {
      */
     public track(
         data: ContractsModule.Contracts.Data<ContractsModule.Contracts.Domain>,
-        tagOverrides?: { [key: string]: string; }) {
+        tagOverrides?: { [key: string]: string; },
+        contextObject?: any) {
```

Comment:
I have looked extensively into async context propagation libraries, for potential use for request correlation in this SDK among other things. Unfortunately there are *no good solutions* available now or in the near future. I think if we were to try to implement it in the SDK it would result in too many issues and a big support burden, to the extent that it wouldn't be worthwhile. Basically there is no way to make it reliable unless/until it is a JavaScript standard. My team is pursuing that angle, but that's a very long-term thing.

In this case, we are able to explicitly pass a context object, as you have done, because there is no user code in the stack between `track()` and the calls to the processors, so there's no need to rely on a library to implicitly forward the context.